components/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/service/SimulationConfigDeployer.java

```diff
@@ -34,39 +34,69 @@
+import org.wso2.carbon.stream.processor.core.internal.StreamProcessorDeployer;
 import org.wso2.carbon.utils.Utils;
 
 import java.io.File;
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.nio.file.Paths;
+import java.util.Collections;
 
 /**
  * SimulationConfigDeployer is responsible for all simulation config deployment tasks
- *
  */
 @Component(
         name = "simulation-config-deployer",
         immediate = true,
         service = org.wso2.carbon.deployment.engine.Deployer.class
 )
-public class SimulationConfigDeployer implements Deployer {
+public class SimulationConfigDeployer implements Deployer, DeployerListener {
     private static final Logger log = LoggerFactory.getLogger(SimulationConfigDeployer.class);
     private ArtifactType artifactType = new ArtifactType<>(EventSimulatorConstants.SIMULATION_FILE_EXTENSION);
     private URL directoryLocation;
 
+    /**
+     * deployConfigFile() is used to deploy a simulation configuration added to directory 'simulation-configs'
+     *
+     * @param file simulation config file added
+     */
     private void deployConfigFile(File file) throws Exception {
         if (!file.getName().startsWith(".")) {
             if (FilenameUtils.isExtension(file.getName(), EventSimulatorConstants.SIMULATION_FILE_EXTENSION)) {
                 String simulationName = FilenameUtils.getBaseName(file.getName());
-                String simulationConfig = SimulationConfigUploader.getConfigUploader().getSimulationConfig
-                        (simulationName, (Paths.get(Utils.getCarbonHome().toString(),
-                                EventSimulatorConstants.DIRECTORY_DEPLOYMENT,
-                                EventSimulatorConstants.DIRECTORY_SIMULATION_CONFIGS)).toString());
-                if (!simulationConfig.isEmpty()) {
-                    EventSimulator simulator = new EventSimulator(simulationName, simulationConfig);
-                    EventSimulationMap.getSimulatorMap().put(simulationName, simulator);
+                try {
+                    String simulationConfig = SimulationConfigUploader.getConfigUploader().getSimulationConfig
+                            (simulationName, (Paths.get(Utils.getCarbonHome().toString(),
+                                    EventSimulatorConstants.DIRECTORY_DEPLOYMENT,
+                                    EventSimulatorConstants.DIRECTORY_SIMULATION_CONFIGS)).toString());
+                    if (!simulationConfig.isEmpty()) {
+                        EventSimulator eventSimulator = new EventSimulator(simulationName, simulationConfig);
+                        EventSimulatorMap.getInstance().getActiveSimulatorMap().put(simulationName,
+                                Collections.singletonMap(eventSimulator, simulationConfig));
+                        log.info("Successfully deployed active simulation '" + simulationName + "'.");
+                    }
+                } catch (ResourceNotFoundException e) {
```

Comment:
I will schedule a code review for this so that we could improve on the implementation. And i will send a new PR for that.